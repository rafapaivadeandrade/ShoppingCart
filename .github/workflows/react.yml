# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs

name: React CI/CD

on:
  push:
    branches: [ "deploy" ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14.x]
        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/

    steps:
    - uses: actions/checkout@v3
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v3
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'npm'
    - run: npm i
    - run: npm test
    - run: npm run build
    env:
          CI: false
      - name: Copy build files
        run: mv ./build ./release/build
      - name: release content
        run: pwd && ls -la ./release
      - name: Archive Production Artifact
        uses: actions/upload-artifact@master
        with:
          name: build
          path: ./release