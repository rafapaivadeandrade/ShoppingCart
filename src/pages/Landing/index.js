import React, { useState, useEffect } from 'react';
import {
  Container,
  Nav,
  FirstNav,
  FirstSpan,
  SecondSpan,
  SecondNav,
  ThirdSpan,
  FourthSpan,
  Main,
  Section,
  SortByLabel,
  SortByDiv,
  Price,
  Popularidade,
  OrdemAlfabetica,
  Box,
  Items,
  Buttons,
  AddButton,
  Aside,
  CartContainer,
  CartTitle,
  Total,
  TotalLabel,
  CartItems,
  CartBorder,
  SaveCart,
  SaveCartLabel,
  Footer,
  Image,
  LabelRow,
  ItemRow,
  Column,
  Label,
  Item,
  CartImage,
  CartDetails,
  CartName,
  CartPrice,
} from './styles';
import { useProduct } from '../../hooks/ContextApi';
import cartIcon from '../../assets/images/icons/cart-icon.svg';

function Landing() {
  const {
    products,
    fetchData,
    sortPrice,
    sortPopularity,
    sortAlphabeticalOrder,
    addToCart,
    cart,
    removeFromCart,
  } = useProduct();
  useEffect(() => {
    fetchData();
  }, []);

  return (
    <Container>
      <Nav>
        <FirstNav>
          <>
            <FirstSpan>Logo</FirstSpan>
          </>
          <>
            <SecondSpan>Home</SecondSpan>
          </>
        </FirstNav>
        <SecondNav>
          <>
            <ThirdSpan>Conta</ThirdSpan>
          </>
          <>
            <FourthSpan>
              Log out <i className="fas fa-arrow-right fa-sm"></i>
            </FourthSpan>
          </>
        </SecondNav>
      </Nav>
      <Main>
        <Section>
          <p>
            <strong>Produtos</strong>
          </p>

          <SortByLabel>Sortear por:</SortByLabel>
          <SortByDiv>
            <Price
              onClick={() => {
                sortPrice();
              }}
            >
              Preço
            </Price>
            <Popularidade onClick={() => sortPopularity()}>
              Popularidade
            </Popularidade>
            <OrdemAlfabetica onClick={() => sortAlphabeticalOrder()}>
              Ordem Alfabética
            </OrdemAlfabetica>
          </SortByDiv>
          <Box>
            {products.map(
              (product) => {
                return (
                  <Items key={product.id}>
                    <Image
                      src={
                        require(`../../assets/images/${product.image}`).default
                      }
                    />
                    <Column>
                      <LabelRow>
                        <Label>Name:</Label>
                        <Label>Price:</Label>
                        <Label>Score:</Label>
                      </LabelRow>
                      <ItemRow>
                        <Item style={{ width: '30px' }}>{product.name}</Item>
                        <Item>R$:{product.price}</Item>
                        <Item>{product.score}</Item>
                      </ItemRow>
                      <AddButton
                        onClick={() => {
                          addToCart(product);
                        }}
                      >
                        <CartImage src={cartIcon} />{' '}
                        <span>Adicionar ao Cart</span>
                      </AddButton>
                    </Column>
                  </Items>
                );
              },
              [products]
            )}
          </Box>
        </Section>
        <Aside>
          <CartContainer>
            <CartTitle>
              <em>CART</em>
            </CartTitle>
            {cart.map((cartItem) => {
              return (
                <CartItems>
                  <i
                    className="far fa-trash-alt"
                    onClick={() => {
                      removeFromCart(cartItem);
                    }}
                  ></i>
                  <CartBorder></CartBorder>
                  <CartDetails>
                    <CartName>
                      <Item style={{ width: '180px' }}>{cartItem.name}</Item>
                    </CartName>
                    <CartPrice>
                      <Item>R$:{cartItem.price}</Item>
                    </CartPrice>
                  </CartDetails>
                </CartItems>
              );
            })}
          </CartContainer>
          <SaveCart>
            <Total>
              <strong>
                <em>CART</em>
              </strong>{' '}
              <TotalLabel>
                Frete: <span></span>{' '}
              </TotalLabel>
            </Total>
            <Total>
              <strong>
                <em>CART</em>
              </strong>{' '}
              <TotalLabel>
                Subtotal: <span></span>{' '}
              </TotalLabel>
            </Total>
            <Total>
              <strong>
                <em>CART</em>
              </strong>{' '}
              <TotalLabel>
                TOTAL: <span></span>{' '}
              </TotalLabel>
            </Total>
          </SaveCart>
        </Aside>
      </Main>
      <Footer>
        <p>Copyright 2021 Supera Tecnologia</p>
      </Footer>
    </Container>
  );
}

export default Landing;
