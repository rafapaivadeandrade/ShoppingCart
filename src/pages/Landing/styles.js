import styled from 'styled-components';

export const Container = styled.div`
  width: 100vw;
`;
export const Nav = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  border: 1px solid #e0dde5;
  height: 55px;
`;
export const FirstNav = styled.nav`
  margin: 10px 0;
  display: flex;
  justify-content: space-evenly;
  width: 350px;
  align-items: center;
  margin-left: -200px;
`;
export const FirstSpan = styled.span`
  font-size: 35px;
  font-weight: bold;
  color: #737373;
  font-style: italic;

  &:hover {
    cursor: pointer;
    border-bottom: 5px solid #b5c401;
    padding-bottom: 5px;
    border-radius: 5px;
  }
`;
export const SecondSpan = styled.span`
  font-size: 15px;
  font-weight: bold;
  color: #737373;
  font-style: italic;
  &:hover {
    cursor: pointer;
    border-bottom: 5px solid #b5c401;
    padding-bottom: 5px;
    border-radius: 5px;
  }
`;
export const SecondNav = styled.nav`
  display: flex;
  justify-content: space-around;
  width: 200px;
  align-items: center;
  margin-right: -100px;
`;
export const ThirdSpan = styled.span`
  font-size: 15px;
  font-weight: bold;
  color: #737373;
  font-style: italic;

  &:hover {
    cursor: pointer;
    border-bottom: 5px solid #b5c401;
    padding-bottom: 5px;
    border-radius: 5px;
  }
`;
export const FourthSpan = styled.span`
  font-size: 15px;
  font-weight: bold;
  color: #737373;
  font-style: italic;
  &:hover {
    cursor: pointer;
    border-bottom: 5px solid #b5c401;
    padding-bottom: 5px;
    border-radius: 5px;
  }
`;
export const Main = styled.main`
  margin-top: 40px;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  height: 80%;
`;

export const Section = styled.section`
  width: 40%;
  display: flex;
  flex-direction: column;
  p:nth-of-type(1) {
    padding-top: 20px;
    color: #737373;
    font-style: italic;
    font-size: 20px;
  }
`;
export const SortByLabel = styled.span`
  font-weight: bold;
  padding-top: 20px;
  color: #737373;
  font-style: italic;
  font-size: 12px;
`;
export const SortByDiv = styled.div`
  padding-top: 10px;
  display: flex;
  flex-direction: row;
  width: 450px;
  justify-content: space-between;
`;
export const Price = styled.div`
  outline: none;
  border-radius: 20px;
  padding: 5px 20px;
  border: 1px solid #7f3992;
  color: #7f3992;
  font-weight: bold;
  font-style: italic;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
    background-color: #7f3992;
    color: #f7f7f7;
  }
`;
export const Popularidade = styled.div`
  outline: none;
  border-radius: 20px;
  padding: 5px 20px;
  border: 1px solid #28c383;
  background-color: #28c383;
  color: white;
  font-weight: bold;
  font-style: italic;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
    background-color: #f7f7f7;
    color: #28c383;
  }
`;
export const OrdemAlfabetica = styled.div`
  outline: none;
  border-radius: 20px;
  padding: 5px 20px;
  border: 1px solid #f9ba70;
  color: #f9ba70;
  font-weight: bold;
  font-style: italic;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
    background-color: #f9ba70;
    color: #f7f7f7;
  }
`;

export const Box = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  width: 80%;
  margin-top: 10px;
`;
export const Items = styled.div`
  box-shadow: 0 2px 7px 1px rgba(0, 0, 0, 0.2);
  width: 450px;
  height: 200px;
  display: flex;
  align-items: center;
  font-size: 30px;
  padding: 15px;
  margin-top: 10px;
  flex-direction: row;
`;
export const Buttons = styled.div`
  border-radius: 20px;
  padding-top: 20px;
  color: #7f3992;
`;
export const AddButton = styled.button`
  color: #ffffff;
  background-color: #27c383;
  outline: none;
  border: 1px solid #27c383;
  font-size: 12px;
  font-weight: bold;
  border-radius: 8px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 200px;
  transition: background-color 1s;
  margin-left: 120px;
  margin-top: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  &:hover {
    opacity: 0.8;
    background-color: transparent;
    color: #28c383;
    cursor: pointer;
    svg {
      opacity: 0.8;
      background-color: transparent;
      color: #28c383;
    }
  }
`;
export const CartImage = styled.img`
  width: 20px;
  height: 20px;
  margin-right: 10px;
`;
export const Aside = styled.aside`
  width: 300px;
  height: 400px;
  border-radius: 8px;
`;
export const CartContainer = styled.div`
  border-top-right-radius: 8px;
  border-top-left-radius: 8px;
  background-color: #ffffff;
  padding: 20px;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  border: 1px solid #e0dde5;
  overflow-y: auto;
`;
export const CartTitle = styled.p`
  margin-top: 10px;
  font-size: 20px;
  font-weight: bold;
  color: #737373;
`;
export const Total = styled.div`
  font-size: 20px;
  color: #737373;
  display: flex;
  flex-direction: row;
`;
export const TotalLabel = styled.p`
  text-indent: 5px;
  font-size: 20px;
`;

export const SaveCart = styled.div`
  border-bottom-left-radius: 8px;
  border-bottom-right-radius: 8px;
  background-color: #f4f4f4;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 40px;
  border: 1px solid #e0dde5;
  height: 24%;
  font-weight: bold;
  align-items: center;
`;
export const SaveCartLabel = styled.p`
  color: #28c383;
  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }
`;
export const Footer = styled.footer`
  border-top: 1px solid #e0dde5;
  position: relative;
  display: flex;
  justify-content: center;
  color: #737373;
  font-size: 15px;
  margin-top: 50px;
  p {
    padding: 20px 0;
  }
`;
export const Image = styled.img`
  width: 100px;
  height: 100px;
  object-fit: cover;
`;
export const LabelRow = styled.div`
  display: flex;
  width: 300px;
  height: 50px;
  flex-direction: row;
  justify-content: space-between;
  margin-left: 10px;
  /* margin-top: -50px; */
`;
export const ItemRow = styled.div`
  display: flex;
  margin-left: 10px;
  width: 300px;
  height: 20px;
  flex-direction: row;
  justify-content: space-between;
  /* margin-top: -20px; */
`;
export const Column = styled.div`
  display: flex;
  flex-direction: column;
  height: 100px;
`;
export const Label = styled.span`
  font-size: 15px;
  color: #737373;
`;
export const Item = styled.span`
  font-size: 15px;
  color: #737373;
`;
export const CartDetails = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 50px;
`;
export const CartName = styled.div`
  display: flex;
  flex-direction: row;
`;
export const CartPrice = styled.div`
  display: flex;
  flex-direction: row;
`;
export const CartItems = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 220px;
  height: 90px;
  margin-bottom: 5px;
  i {
    font-size: 15px;
    cursor: pointer;
  }
`;
export const CartBorder = styled.div`
  height: 60px;
  width: 3px;
  background-color: #01ac66;
`;
