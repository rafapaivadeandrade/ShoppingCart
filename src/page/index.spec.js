import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Landing from './index';
import createStore from '../redux/index';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';

describe('landing', () => {
  const store = createStore;

  it('should render Landing Component', () => {
    const result = render(
      <Provider store={store}>
        <Landing />
      </Provider>
    );
    expect(result).toBeTruthy();
  });
});
