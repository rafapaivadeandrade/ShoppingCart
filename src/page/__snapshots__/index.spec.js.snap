// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing should render Landing Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ bJeKWn"
      >
        <nav
          class="sc-gsTCUz cKRcxm"
        >
          <nav
            class="sc-dlfnbm fxihgT"
          >
            <span
              class="sc-hKgILt hnuySZ"
            >
              Shopping
            </span>
          </nav>
          <nav
            class="sc-eCssSg kovFzu"
          >
            <div
              style="width: 40px; height: 40px; border-radius: 25px; background-color: rgb(102, 102, 102);"
            />
             
            <span
              class="sc-jSgupP gNaAFZ"
            >
              John doe
            </span>
          </nav>
        </nav>
        <main
          class="sc-gKsewC jLVesz"
        >
          <section
            class="sc-iBPRYJ dYemLy"
          >
            <div
              class="sc-pFZIQ iBhVVZ"
            />
          </section>
          <aside
            class="sc-kstrdz cRLqoW"
          >
            <p
              class="sc-fFubgz bTuLHK"
            >
              Shopping Cart
            </p>
            <div
              class="sc-hBEYos jTeIkF"
            />
            <div
              class="sc-fodVxV hVYtmE"
            >
              <input
                aria-label="Discount"
                class="sc-bYEvPH iolMaG"
                placeholder="Discount code"
                value=""
              />
              <button
                class="sc-kLgntA esJTBh"
              >
                APPLY
              </button>
            </div>
            <div
              class="sc-dIUggk jUzZkq"
            >
              <p
                class="sc-idOhPF gTcBvD"
              >
                Subtotal
                <span>
                  $ 0.00
                </span>
                 
              </p>
              <p
                class="sc-idOhPF gTcBvD"
              >
                Shipping:
                <span>
                  $ 0.00
                </span>
                 
              </p>
              <p
                class="sc-idOhPF gTcBvD"
              >
                Discount:
                <span>
                  $ 0.00
                </span>
                 
              </p>
              <div
                class="sc-bkzZxe dOJPnG"
              >
                <p
                  class="sc-idOhPF gTcBvD"
                  style="color: rgb(51, 51, 51);"
                >
                  Total:
                  <span>
                    $ 0.00
                  </span>
                   
                </p>
              </div>
            </div>
            <button
              class="sc-iktFzd hxXOri"
            >
              CHECKOUT
            </button>
          </aside>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ bJeKWn"
    >
      <nav
        class="sc-gsTCUz cKRcxm"
      >
        <nav
          class="sc-dlfnbm fxihgT"
        >
          <span
            class="sc-hKgILt hnuySZ"
          >
            Shopping
          </span>
        </nav>
        <nav
          class="sc-eCssSg kovFzu"
        >
          <div
            style="width: 40px; height: 40px; border-radius: 25px; background-color: rgb(102, 102, 102);"
          />
           
          <span
            class="sc-jSgupP gNaAFZ"
          >
            John doe
          </span>
        </nav>
      </nav>
      <main
        class="sc-gKsewC jLVesz"
      >
        <section
          class="sc-iBPRYJ dYemLy"
        >
          <div
            class="sc-pFZIQ iBhVVZ"
          />
        </section>
        <aside
          class="sc-kstrdz cRLqoW"
        >
          <p
            class="sc-fFubgz bTuLHK"
          >
            Shopping Cart
          </p>
          <div
            class="sc-hBEYos jTeIkF"
          />
          <div
            class="sc-fodVxV hVYtmE"
          >
            <input
              aria-label="Discount"
              class="sc-bYEvPH iolMaG"
              placeholder="Discount code"
              value=""
            />
            <button
              class="sc-kLgntA esJTBh"
            >
              APPLY
            </button>
          </div>
          <div
            class="sc-dIUggk jUzZkq"
          >
            <p
              class="sc-idOhPF gTcBvD"
            >
              Subtotal
              <span>
                $ 0.00
              </span>
               
            </p>
            <p
              class="sc-idOhPF gTcBvD"
            >
              Shipping:
              <span>
                $ 0.00
              </span>
               
            </p>
            <p
              class="sc-idOhPF gTcBvD"
            >
              Discount:
              <span>
                $ 0.00
              </span>
               
            </p>
            <div
              class="sc-bkzZxe dOJPnG"
            >
              <p
                class="sc-idOhPF gTcBvD"
                style="color: rgb(51, 51, 51);"
              >
                Total:
                <span>
                  $ 0.00
                </span>
                 
              </p>
            </div>
          </div>
          <button
            class="sc-iktFzd hxXOri"
          >
            CHECKOUT
          </button>
        </aside>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
